### PR TITLE
Blog - Working on ESP devkits using WSL2

### PR DESCRIPTION
**Pull request by jakub-kocka**
## Description

* Introduces the topic of working with Espressif's devkits using the WSL
* Simple guide on how to use the WSL with serial port forwarding

## Related

* Internal tracker IDF-4361

## Testing

2024-10-15 15 42 36 localhost 8327fde99f8a

### CODE REVIEW

**Comment by mfialaf** [comment]
Things that may be worth mentioning:
- In step 2, the `usbipd` can be seen only after opening new terminal
- Step `usbipd bind --busid 1-1` requires Administrator privileges 

I've successfully managed to complete the tutorial. I can see, the new device using the `susb` command in the WSL terminal.
But when listing the ports `ls /dev/tty*`, there is no change. I assumed that I can use this article to communicate with the chip from wsl, but that require the port specification.

Am I missing something? Is this the point where I need to use the `esp_rfc2217_server`?

**Comment by jakub-kocka** [comment]
Thank you, @mfialaf for the review.

I have improved the description of the steps you have pointed out. Also updated the guide with the mentioned issue as we have solved this in person. Thank you for the help!

File: content/blog/espressif-devkits-with-wsl2/index.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+title: "Working with Espressif's devkits using WSL2"
+date: 2024-10-10
```
Comment (pedrominatel) [inline content/blog/espressif-devkits-with-wsl2/index.md:3]: Update to a recent date.

**Comment by pedrominatel** [comment]
Hi @jakub-kocka, thank you for the article.

Can you please rebase? This will fix some issues and the CI will pass. Also, update the article date to a recent one.

**Comment by jakub-kocka** [comment]
Hi @pedrominatel, thanks for the review. However, rebase doesn't seem to help.

**Comment by pedrominatel** [comment]
Don't worry, @jakub-kocka . I'll take a look.

File: content/blog/espressif-devkits-with-wsl2/index.md [hunk 2]
```diff
@@ -1,77 +1,103 @@
...
+
 ## Introduction
 
-[WSL](https://learn.microsoft.com/en-us/windows/wsl/) (Windows Subsystem for Linux) is a great way to run Linux alongside Windows. However, when there is a need to use it with Espressif's devkits, it can be a bit tricky due to the required serial port forwarding. In this short article, a step-by-step guide is provided to help you work with Espressif's devkits using WSL.
+Developing ESP32's based software solutions with Windows can be, sometimes, overwhelming due to some characteristics of the OS that increases the build time in some configurations or security polices.
+
+To overcome, the WSL is an good option to optimize, specially the build time. However, when there is a need to use it with Espressif's SoCs with USB-CDC or external USB-to-Serial (`CP210x`, etc), it can be a bit more tricky due to the required serial port forwarding.
```
Comment (jakub-kocka) [inline content/blog/espressif-devkits-with-wsl2/index.md:16]: I am not sure if we should mention the build time here. For example, in my case using WSL is significantly (e.g. at least 5 times) slower than using a PowerShell console with an IDF environment.

This is also why I was indirect about the use case. We have seen these issues when users want to use SoCs with WSL, so this article exists, but to be honest, I am not certain about a clear and good use case.

**Review by jakub-kocka — COMMENTED** [review]
Thank you @pedrominatel for the review and changes, it looks good to me.
I have left just a note about one sentence.

File: content/blog/espressif-devkits-with-wsl2/index.md (hunk 2 above)
Comment (pedrominatel) [reply content/blog/espressif-devkits-with-wsl2/index.md]: Good point! PTAL at the updated version, @jakub-kocka.

Comment (jakub-kocka) [reply content/blog/espressif-devkits-with-wsl2/index.md]: Nice improvement, thank you @pedrominatel. LGMT!

**Comment by f-hollow** [comment]
Not sure what is wrong with the product selector link. It keeps appearing as invalid in this PR. However, as soon as I push these changes to a [test PR](https://github.com/espressif/developer-portal/pull/178), the checks always pass successfully and the status of checks here immediately switches to all passed.

This [PR](https://github.com/espressif/developer-portal/pull/180) should fix such issues, hopefully, once and for all!

**Pull request merged**